schema/table.go

```diff
@@ -163,7 +157,7 @@ func (t Table) TableNames() []string {
 }
 
 // Call the table resolver with with all of it's relation for every reolved resource
-func (t Table) Resolve(ctx context.Context, meta ClientMeta, parent *Resource, resourcesSem *semaphore.Weighted, resolvedResources chan<- *Resource) (summary SyncSummary) {
+func (t *Table) Resolve(ctx context.Context, meta ClientMeta, parent *Resource, resourcesSem *semaphore.Weighted, resolvedResources chan<- *Resource) (summary SyncSummary) {
```

Comment:
Im not sure we really want to turn it into `*` at least not in this PR or why it is needed at all?

Reply:
https://go.dev/doc/faq#methods_on_values_or_pointers

The pointer receiver is required because I added a method that changes the table (if the receiver is a value, it is impossible to have methods that modify the `Table`).

From the FAQ I linked:

> Next is consistency. If some of the methods of the type must have pointer receivers, the rest should too, so the method set is consistent regardless of how the type is used. See the section on [method sets](https://go.dev/doc/faq#different_method_sets) for details.

And also 

> Second is the consideration of efficiency. If the receiver is large, a big struct for instance, it will be much cheaper to use a pointer receiver.

The table struct is very large..

Unless there is a specific reason to use value-receiver (I can't think of one here), than pointer-receivers are a good default.